src/scrapping.ts

```diff
@@ -39,28 +39,38 @@ function extractVersion(input: string) {
   return match ? match[0] : undefined;
 }
 
-export async function getStableLatestVersion(org: string, repo: string) {
+export async function getVersion(org: string, repo: string, ver: string) {
   const apkmUrl = `https://www.apkmirror.com/apk/${org}/${repo}`;
 
   const response = await fetch(apkmUrl);
   const html = await response.text();
   const $ = cheerio.load(html);
 
+  if (await response.status != 200) {
+    throw new Error("Got response code: " + response.status);
+  }
+
   const versions = $(
     `#primary > div.listWidget.p-relative > div > div.appRow > div > div:nth-child(2) > div > h5 > a`
   )
     .toArray()
     .map((v) => $(v).text());
 
-  const stableVersion = versions.filter(
-    (v) => !v.includes("alpha") && !v.includes("beta")
-  )[0];
+  var selectedVersion;
 
-  if (!stableVersion) {
-    throw new Error("Could not find stable version");
+  if (!ver) {
+    selectedVersion = versions.filter(
+      (v) => !v.includes("alpha") && !v.includes("beta")
+    );
+  } else {
+    selectedVersion = versions.filter((v) => v.includes(ver));
```

Comment:
Since we used the name `selectedVersion` instead of `selectedVersions`, it's not expected to have an array.


```suggestion
  if (!ver) {
    selectedVersion = versions.filter(
      (v) => !v.includes("alpha") && !v.includes("beta")[0]
    );
  } else {
    selectedVersion = versions.filter((v) => v.includes(ver))[0];
```